### PR TITLE
Components: refactor `ComboboxControl` to pass `exhaustive-deps`

### PR DESCRIPTION
**Pull request by chad1008**
## What?
Updates the `ComboboxControl` component to satisfy the `exhaustive-deps` eslint rule

## Why?
Part of the effort in https://github.com/WordPress/gutenberg/pull/41166 to apply `exhuastive-deps` to the Components package

## How?

I think we can safely remove `value` from the `matchingSuggestions` `useMemo` dependency array. `value` is a prop that, as far as I can tell, doesn't impact that matches based on current input.

The main risk here felt like the impact of the parent component updating `value` on us mid-search, so I've tested this in Storybook by adding an input that fires `setValue` on a five second delay. During that delay I began typing into the ComboBox and waiting for the `value` to change causing a rerender. When it did, the current matches didn't appear to be disrupted, and there was no impact in selecting an option based on the current input.

Just in case, @youknowriad, do you have any recollection of the details on adding `value` to this dependency array? It was a long time ago, but I wanted to double check in case I've overlooked something :) 

## Testing Instructions
1. From your local Gutenberg directory, run `npx eslint --rule 'react-hooks/exhaustive-deps: warn' packages/components/src/combobox-control`
2. Confirm that the linter returns no errors
3. Launch Storybook, test `ComboboxControl`'s stories/docs to ensure they still work as expected with no console errors
4. Confirm e2e tests still pass

### CODE REVIEW

**Review by ciampo — APPROVED** [review]
Thank you @chad1008 for working on this!

This component, in its current "hook" form, was first added in #25442 — including the `useMemo` hook in question and `value` in the list of dependencies. Skimming through the PR, I couldn't find any particular mention justifying `value` as an extra dependency.

The **lack of unit tests** for this component give us less confidence when making this sort of changes, but the code changes LGTM at a first glance. I also had a look at Storybook and I the component seems to work as expected.

I'm going to approve this PR, but in the meantime let's see if @youknowriad has anything to add to clarify your doubts.

**Comment by chad1008** [comment]
I've just done another pass of manual testing in Storybook and everything appears to be behaving as expected with this change. @ciampo what are your thoughts at this point? Should we merge, or wait for feedback from @youknowriad?

**Review by ciampo — COMMENTED** [review]
> @ciampo what are your thoughts at this point? Should we merge, or wait for feedback from @youknowriad?

@youknowriad  won't be around for the next couple of months.

I would probably give a final round of testing in the editor, before merging, and then I'd keep an eye open for any potential bug reports on UIs using this component.

Alternatively, depending on how you feel about it and your availability, we could also open a PR to add a few unit tests to this component, in order to gain more confidence for the changes in this PR — up to you, both options are ok to me.

File: packages/components/CHANGELOG.md
Comment (stokesman) [inline packages/components/CHANGELOG.md]:
```suggestion
-   `ComboboxControl` updated to satisfy `react/exhaustive-deps` eslint rule ([#41417](https://github.com/WordPress/gutenberg/pull/41417))
```
à la #42244 👨‍🎨

Comment (chad1008) [reply packages/components/CHANGELOG.md]: Thanks for the fix!

**Comment by chad1008** [comment]
First pass at some unit tests in [this draft](https://github.com/WordPress/gutenberg/pull/42403). It doesn't have everything I'd like, and I'm open to input on what more I should add! For now though, this PR is passing those proposed tests :)

**Comment by ciampo** [comment]
> First pass at some unit tests in [this draft](https://github.com/WordPress/gutenberg/pull/42403). It doesn't have everything I'd like, and I'm open to input on what more I should add! For now though, this PR is passing those proposed tests :)

Awesome. Once #42403 is merged, we can rebase this PR and merge it too in case all unit tests are ✅

**Comment by chad1008** [comment]
The new unit tests are merged, and passing locally after a rebase. Letting the tests run here in GH and then we can merge this as well!

**Pull request merged**
